### ISSUE TITLE
Azure Active Directory Authentication issue

### ISSUE THREAD

**Issue by PhilipCotan** [Requarks/wiki#307]
### Issue steps

After setting up AAD Auth, login doesn't work because the system cannot find the user email in the json object returned from Azure. 

### Fix
In order to fix this issue, in **./server/models/user.js** I had to make the following changes on lines 54-55:

<p><i><strong>} else if (_.isString(profile.unique_name) && profile.unique_name.length > 5) {  
<br>primaryEmail = profile.unique_name </strong></i></p>

Because in the Azure response object, the user email is stored in **unique_name** property instead of _email, emails or mail_.

**Comment by barry-luijten** [comment]
Inserting these lines before line #61 of server/models/user.js:
```
 } else if (profile.unique_name && profile.unique_name.length > 5) {
    primaryEmail = profile.unique_name
```
And changing line 77 to:
```name: profile.displayName || profile.name || profile.cn || _.split(primaryEmail, '@')[0]```
... fixed the issue for me....

**Comment by barry-luijten** [comment]
Bug still exists in version 1.0.68... Please fix this, it's about 2 minutes work ;)

**Comment by NGPixel** [comment]
Fixed in 1.0.78.

**Comment by ekaragic** [comment]
Bug is not fixed in version 1.0.78.

**Comment by ekaragic** [comment]
Confirmed that issue is fixed.
